app/components/documents/document-card/edit-modal.js

```diff
@@ -69,22 +69,24 @@ export default class DocumentsDocumentCardEditModalComponent extends Component {
       yield oldFile.destroyRecord();
       this.args.piece.file = this.replacementFile;
```

Comment:
We need to relink the existing derived file (if any) to the replacement source file here